application/managers/http/manager.py

```diff
@@ -0,0 +1,76 @@
+"""
+Manager for http business logic.
+"""
+
+
+from fastapi import Depends
+
+from constants.events import EventCategory
+from managers.events import EventManager
+from managers.events import get_event_manager
+from managers.kubernetes import K8sManager
+from managers.organizations.manager import OrganizationManager
+from managers.organizations.manager import get_organization_manager
+from models.organization import Organization
+from schemas.events import EventSchema
+from services.helm.facade import HelmService
+import httpx
+
+
+
+class HttpManager:
+    """
+    Manager to manipulate Http requests.
+    """
+    organization_manager: OrganizationManager
+
+    def __init__(self, organization_manager: OrganizationManager, event_manager: EventManager):
+        self.organization_manager = organization_manager
+        self.event_manager = event_manager
+
+
+    ############################################################################
+    # Http Requests
+    ############################################################################
+
+
+    async def http_request(
+        self, organization: Organization, context_name: str, component_name: str, url: str, method: str,
+        parameters: dict, headers: dict = None, dry_run: bool = False
+    ) -> str:
+        """
+        Send an HTTP request to the specified URL.
+        """
+        # Convert to a dict if a list of dicts provided
+        if isinstance(parameters, list):
+            parameters = {k: v for d in parameters for k, v in d.items()}
+
+        # Create an HTTP client
+        async with httpx.AsyncClient() as client:
+            # Prepare the request
+            if method.lower() == 'get':
```

Comment:
It is working but where is a cute match case?
Also we have [HTTPMethods](https://github.com/JovianX/Service-Hub/blob/c6e80eb9e6a74462ba14ae036252407d66e96982/application/constants/common.py#L10C7-L10C18) enum for http methods. But furthermore you have defined [httpMethods](https://github.com/JovianX/Service-Hub/pull/283/files#diff-3716243d060a9d4e5f05705851171a492c3b8f834f1a7f58ff4611f41a372527R63) and now comparing it to bare stings.

Reply:
I renamed httpMethods to ComponentTypeHttpMethods to better represen that these types are for componentHttp.